### DISCUSSION TITLE
Follow up on supporting Immich announcement - change of wording

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11226] [Announcements]
# Follow up on supporting Immich announcement

Hello everybody,

Alex from Immich here. What a controversy that we caused with the choice of wording, right? My personal apology to you all.

On behalf of the team, we would like everyone to know that we hear your concerns and we appreciate the love and care that you all have expressed for the project. At the end of the day, what we want most is to make sure you are all happy using the app.

With that said, we are working on a change to the word `license`: we will not call it `licensed` or `unlicensed` anymore. What will it be called?

We are still thinking of different options to make the wording less confusing. The new wording will hopefully showcase our intentions properly going forward.

We’re also working on updating the FAQ with more information to clarify those intentions. We just want to provide good software that people will want to pay for whilst not limiting your usage in any way if you can’t.

So expect these changes over the next week. We’re pushing this out now to let you know our appreciation for the feedback you’ve given us.

The amount of purchases in the first 24 hours has been overwhelming. Thanks everyone for the great support!

Have a great weekend!

Immich team.

**Comment by othyn** [comment]
Thank you for taking the feedback to heart and working with the community, this is the Immich team we all love to see. Looking forward to what the Immich team come up with as a revision. Have a great weekend too!

**Comment by joshhoughton** [comment]
Much more reassured. Like others have suggested, `Supporter ❤️` would do nicely

**Reply by chrisjgray** [reply]
Supporter makes the most sense. The amount someone chooses to support shouldn't really matter. You could even Tier it if you really wanted to be fancy and have ranks of support.

**Reply by Chainedge** [reply]
This is probably the the most straightforward one. Can't go wrong with `supporter` when it's a donation, compared to `licensed` where it sounds like there's the expectancy of the end user paying for a license to use the full program, just like most paid-for products.

**Reply by droopified** [reply]
i vote "supporter" too. this would cause less confusion on the differences of what it actually is.

**Reply by lubor-slusny** [reply]
I'd suggest unlicensed->freeloader, licensed->supporter

**Reply by ghost** [reply]
> I'd suggest unlicensed->freeloader, licensed->supporter

Yeah, just insult people before they can even support the project. Sounds like a great idea. 🙄

**Reply by K3das** [reply]
I think it's still cool if there's a message urging people to support the project, so instead of "unlicensed", the message could look something like this:

![image](https://github.com/user-attachments/assets/fd1da3da-42b7-404e-86e0-3ab361057ddc)

![image](https://github.com/user-attachments/assets/29831e8d-d064-451c-9b20-6ab30e40ba5c)

**Reply by athornfam2** [reply]
> I think it's still cool if there's a message urging people to support the project, so instead of "unlicensed", the message could look something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/20052500/350655927-fd1da3da-42b7-404e-86e0-3ab361057ddc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NDMwNTYsIm5iZiI6MTcyMTQ0Mjc1NiwicGF0aCI6Ii8yMDA1MjUwMC8zNTA2NTU5MjctZmQxZGEzZGEtNDJiNy00MDRlLTg2ZTAtM2FiMzYxMDU3ZGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDAyMzIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NGU5MTMzODU1Yzc1NDlmMDgzNzNlYjJkOGRhYmI5YTcwMTAxYTIxZTQxNTEzNDcxYTVkODcyNDczYzg2MTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.auz4CnbOuAZYOf3aLz3siHShItNMZ1cW9SBQ37c4ltM)

I would rather this. I'm a little undecided where I want its placement though. If you put it on the landing page of the user its just there constantly but if you put it behind in the admin settings it will more than likely not be noticed. Not sure. I wish we could go back to Github because I'd rather donate monthly than commit to 1 amount 24.99 or the 99.99.

**Reply by mertalev** [reply]
> I think it's still cool if there's a message urging people to support the project, so instead of "unlicensed", the message could look something like this:
> 
> ![image](https://private-user-images.githubusercontent.com/20052500/350655927-fd1da3da-42b7-404e-86e0-3ab361057ddc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NDc5MzcsIm5iZiI6MTcyMTQ0NzYzNywicGF0aCI6Ii8yMDA1MjUwMC8zNTA2NTU5MjctZmQxZGEzZGEtNDJiNy00MDRlLTg2ZTAtM2FiMzYxMDU3ZGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDAzNTM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQzMjA4ZmZkYjQyOTg5ZGJkMGMyODdlMDg5ZTRmNDI3NzQ3Y2EwNjExNTRhMzljMDllZWJhNTBlZWQ2MWY1YzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8Muu53RoJrm97u6v-swADfufZPqBW_yFpPcvcycU0Ck)
> 
> ![image](https://private-user-images.githubusercontent.com/20052500/350657540-29831e8d-d064-451c-9b20-6ab30e40ba5c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NDc5MzcsIm5iZiI6MTcyMTQ0NzYzNywicGF0aCI6Ii8yMDA1MjUwMC8zNTA2NTc1NDAtMjk4MzFlOGQtZDA2NC00NTFjLTliMjAtNmFiMzBlNDBiYTVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDAzNTM1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwZjVhOWRlZGVjZTExZGJhYzk0Nzg3Y2Q3YmRjMDM2NzY4ZDZjMmQ1MTYxMjU0MGQ1NjdmNWYzODg2MGIxOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.KguraX0eVU8eVXwPy7xfWfwbt2o05YWIOjQZ6HSKfxU)

I think this might just be my favorite. It has a warm and positive vibe that brings the option to pay to your attention without naming the free version as anything. It also just looks really nice and is a perfect fit for the UI.

**Reply by topmax** [reply]
> > I think it's still cool if there's a message urging people to support the project, so instead of "unlicensed", the message could look something like this:
> > ![image](https://private-user-images.githubusercontent.com/20052500/350655927-fd1da3da-42b7-404e-86e0-3ab361057ddc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NDMwNTYsIm5iZiI6MTcyMTQ0Mjc1NiwicGF0aCI6Ii8yMDA1MjUwMC8zNTA2NTU5MjctZmQxZGEzZGEtNDJiNy00MDRlLTg2ZTAtM2FiMzYxMDU3ZGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDAyMzIzNlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4NGU5MTMzODU1Yzc1NDlmMDgzNzNlYjJkOGRhYmI5YTcwMTAxYTIxZTQxNTEzNDcxYTVkODcyNDczYzg2MTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.auz4CnbOuAZYOf3aLz3siHShItNMZ1cW9SBQ37c4ltM)
> 
> I would rather this. I'm a little undecided where I want its placement though. If you put it on the landing page of the user its just there constantly but if you put it behind in the admin settings it will more than likely not be noticed. Not sure. I wish we could go back to Github because I'd rather donate monthly than commit to 1 amount 24.99 or the 99.99.

I would also prefer to donate monthly instead a one-time donation amount of $99.99. With a one-time payment you are always dependent on new supporters. But what if many have already paid the $99.99?

**Reply by stephen304** [reply]
A recurring donation option would be nice but it might be a lot more complicated for the team to implement if they stick to using the license key mechanism they made - either the recurring keys expire (the valid month/year could be part of the key which is protected by the activation code, allowing offline expiration checking) but you have to get a new one every month/year - or alternatively something has to phone home (browser or server) to update the renewal of the existing key so that the same key can be renewed without the key changing (or you could have the expiration be embedded in the activation by implicitly adding the current M/Y onto the key before generating the activation so that only the activation changes when renewed - that would allow offline expiration checking and allow your server/browser to renew the key automatically by refreshing the activation code by re-sending the expired key). But then you need a payment mechanism to renew a key and then they need to also store and update how long a key is valid (or in the case of using the activation key, updating the activation to the current M/Y when a renewal is made).

For reference there are currently IMSV and IMCL keys for server-wide and individual licenses, so they could do one or more of IMSM, ISMY, IMCM, IMCY for monthly/yearly client/server-wide keys depending on what options they want, with the main difference being that the activation server / local immich server adds the year or year + month before checking it against / generating the activation, while storing just the key so the instance/browser can update the expiration just by re-fetching the activation. Just spitballing...

If they're up for the task I think it would be neat, as long as it's explicitly a donation for a badge. Signal's badges convinced me to donate for a while.

**Reply by tomsik-radek** [reply]
> A recurring donation option would be nice but it might be a lot more complicated for the team to implement if they stick to using the license key mechanism they made - either the recurring keys expire (the valid month/year could be part of the key which is protected by the activation code, allowing offline expiration checking) but you have to get a new one every month/year - or alternatively something has to phone home (browser or server) to update the renewal of the existing key so that the same key can be renewed without the key changing (or you could have the expiration be embedded in the activation by implicitly adding the current M/Y onto the key before generating the activation so that only the activation changes when renewed - that would allow offline expiration checking and allow your server/browser to renew the key automatically by refreshing the activation code by re-sending the expired key). But then you need a payment mechanism to renew a key and then they need to also store and update how long a key is valid (or in the case of using the activation key, updating the activation to the current M/Y when a renewal is made).
> 
> For reference there are currently IMSV and IMCL keys for server-wide and individual licenses, so they could do one or more of IMSM, ISMY, IMCM, IMCY for monthly/yearly client/server-wide keys depending on what options they want, with the main difference being that the activation server / local immich server adds the year or year + month before checking it against / generating the activation, while storing just the key so the instance/browser can update the expiration just by re-fetching the activation. Just spitballing...
> 
> If they're up for the task I think it would be neat, as long as it's explicitly a donation for a badge. Signal's badges convinced me to donate for a while.

Or they could reopen donations again. Or just say "this is for a year" without any enforcement, just like they don't check activation count, it would be up to you to "stop using" when you stop paying.

**Reply by stephen304** [reply]
True, I was mainly thinking if they spent all this effort to make a redemption mechanism for what amounts to a donation for a cosmetic, they might want to use it for reoccurring donations too, and I'm sure some people would want to get a cosmetic for reoccurring donations if that was enabled again. OTOH I'm sure some people also just want to donate and not mess with keys or whatever. I'm not strongly attached to what they decide regarding reoccurring donations / cosmetics.

**Comment by MergenStudios** [comment]
As this is a discussion, may I suggest something along the lines of "Donor" / "Donator" / "Supporter" instead of "Licensed"? I heard it a lot in the discord, and it makes a lot of sense in my oppinion.

**Reply by Solmark** [reply]
Agreed, something that shows that you've supported the Immich development but without it appearing as though you have to, or that you are somehow a second class citizen if you don't.

**Comment by thariq-shanavas** [comment]
I personally love the language used by Signal. I think it is reasonable to show an occasional popup in the app/website to request the user/admin purchase a license. If a server license or individual license has been purchased, we can show a badge next to the User profile picture that says "Sustainer". To avoid nagging users who don't have the ability to purchase a license, maybe show fewer popups as time passes and/or show a popup at most once per minor update.

This is different from the usual donation ask that users happily ignore by putting the licensing model of one-time purchase front and center.

Popup, maybe once every other week:
![image](https://github.com/user-attachments/assets/877a17b3-d96d-4528-a207-9062f60606bb)

Sustainer badges:
![image](https://github.com/user-attachments/assets/2b29dcac-6b1b-4d61-8cd5-877614c64823)

This is a revolutionary way of building software - Thank you for your team, and everyone at FUTO for your work in making open-source sustainable.<3

**Reply by ghost** [reply]
A pop-up would be annoying... don't we have enough pop-ups to deal with these days?

The badges you showed off look really nice though!

**Reply by thariq-shanavas** [reply]
The badges are not an original design, they're just sustainer badges from Signal. I used it as an example here.

I think popups can be less frequent as time goes on if a user has no intention or ability to buy a license, but I think it's less intrusive and more transparent than an 'unlicensed' label, while getting the point across.

**Reply by oddstr13** [reply]
I would like to note, that if your android app has a pop-up asking for store reviews, you're getting a one or two star review from me – the main differentiator being whether I was able to do what I wanted before the popup showed up or not.
Do ***NOT*** get in my way when I'm doing things.

**Reply by alextran1502** [reply]
@oddstr13 I don't like those popups either so don't worry about it

**Reply by oddstr13** [reply]
@alextran1502 I should have noted that it was a general statement, and not directed at any particular project. :)

**Comment by sandersondigital** [comment]
@alextran1502 You shouldn't have to apologise, you have put time and effort into a great application which does not come without its costs and I don't want the project to die.  That's why I've signed up today to 'support' you for my full server.  Keep up the good work and please dont stop looking at a solution for nested albums 😃

**Comment by Wolfhound905** [comment]
**In my opinion**, I do not think the naming needs to change. It's not a donation. You are purchasing a license for your use of the product. There is an infinite trial period until you have the means to purchase a license. There is no need to apologize for any wording. Some people just don't understand how an open soruce project can be paid.

Edit: At the time of this edit, there's 39 dislikes. I would like to mention that this wording is used on apps like Grayjay and Futo Keyboard. I'm unsure what makes people think that my take is bad. A lot of the other opinions I see are just trying to say that they want to turn this back into a donation. The developers said it themselves, they made more money in the past 24 hours of the license change than they made with the donation model.
![Screenshot_20240720-132346](https://github.com/user-attachments/assets/34dbf000-a0d6-41aa-970f-7a7976cc82b8)

**Reply by eigenphase** [reply]
Either you did not read through the comments and concerns in the other thread or you still don't understand it. This is NOT a license! The license is AGPL. If there is a (different) license, there needs to be a proper agreement with a clear definition of what this "license" (a) requires a license holder to do (b) allows a license holder to do.

**Reply by tomsik-radek** [reply]
> **In my opinion**, I do not think the naming needs to change. It's not a donation. You are purchasing a license for your use of the product. There is an infinite trial period until you have the means to purchase a license. There is no need to apologize for any wording. Some people just don't understand how an open soruce project can be paid.
> 
> Edit: At the time of this edit, there's 39 dislikes. I would like to mention that this wording is used on apps like Grayjay and Futo Keyboard. I'm unsure what makes people think that my take is bad. A lot of the other opinions I see are just trying to say that they want to turn this back into a donation. The developers said it themselves, they made more money in the past 24 hours of the license change than they made with the donation model. 

For the record, I don't like this wording on GrayJay and other FUTO apps either.

**Reply by basdp** [reply]
That's not how the AGPL works though.

**Reply by mvdkleijn** [reply]
> That's not how the AGPL works though.

How so? The AGPL is *the* license for this project. You are never required to pay for the license.

By saying you have to pay for a license they're either:

a) implying that you're denied using the software under the AGPL if you don't pay up (which they implied wasn't so since there wouldn't be a requirement to pay

or

b) that there is *another* license applicable... which would make you "licensed" or "unlicensed" depending on whether you pay or not.

Since the case appeared to be b), I think it was/is entirely fair to demand a fully written out license text because you're entering into a legal agreement.

I've seen too many people trying to turn popular projects into commercial products for me not to want proper license texts.

**Comment by rbuchberger** [comment]
Absolutely no apology is needed here. You’re doing great work, and honestly a lot of people in the community owe you an apology. They don’t know the difference between free beer and free speech, and apparently feel entitled to your labor forever. I wish so many didn’t immediately assume the worst intentions based on perceived vibes and not much else. 

I like the idea of avoiding charity related language, but I do have one slight criticism of the word license: it creates confusion with the software license it’s released under. 

Maybe “purchased copy” and “free copy?”

**Reply by jordanlambrecht** [reply]
I don't want a shame-badge saying "free copy" every time I open my instance.

**Reply by jantari** [reply]
@jordanlambrecht then pay up. There's no shame in using something free, but if you don't like it then you have the option to pay.

**Reply by rbuchberger** [reply]
> I don't want a shame-badge saying "free copy" every time I open my instance.

Then pay, or hide it with custom css, or toggle the flag in the db, or fork it and remove the badge. 

The reality is that devs gotta eat, and selling copies of software, even GPL licensed software, is not just ethical but it’s one of the few decent ways to sustainably fund such a project.

**Reply by ghost** [reply]
> The reality is that devs gotta eat, and selling copies of software, even GPL licensed software, is not just ethical but it’s one of the few decent ways to sustainably fund such a project.

Nobody has a problem with their generous pricing model. It’s the language being used.

Purchased copy and free copy is poor word choice as well. Better than licensed and unlicensed though.

`Supporter <3` as mentioned above makes the most sense in my opinion. It does a great job of making you feel good for supporting the project without making you feel bad if you can’t for whatever reason.

Licensed, Unlicensed, and Unlimited Trial are all terms that received backlash for good reason. It wouldn’t take much to move the goal post at that point.

**Reply by colin969** [reply]
A constant immovable reminder to donate would be a real pain for a self hosted service. There are plenty of ways to encourage people to donate that doesn't come across as badgering and taking control from them.

The day I start having to use adblock on self hosted services, just pull the plug honestly.

**Reply by stephen304** [reply]
@colin969 I hope the follow what Futo did with Futo voice/keyboard where you can just press "I paid" and it doesn't check. Or maybe add an env var to hide the donation box (but leave it in the settings so that users can still add their key if they want to). It doesn't seem like Futo are the kind of people to fight and obfuscate to bug us with a banner against our will.

**Reply by rbuchberger** [reply]
> Purchased copy and free copy is poor word choice as well.

But why though? It creates an expectation that you should probably pay for this software, in the gentlest possible way. `Supporter <3` doesn't create that expectation.

**Reply by mvdkleijn** [reply]
> > Purchased copy and free copy is poor word choice as well.
> 
> But why though? It creates an expectation that you should probably pay for this software, in the gentlest possible way. `Supporter <3` doesn't create that expectation.

First of all, there's nothing gentle about it.

Secondly, you're missing the point of open source in my opinion. There is no requirement to pay for the source. That's (one of) the whole point of the AGPL. By giving away your code under the AGPL (and most other open source licenses) there's no (legal) expectation for someone to pay.

Open Source is essentially volunteer work that hopefully grows enough support to make it a full time, sustainable job replacement in a happy scenario.

However, it is still volunteer work at the end of the day. Volunteers are paid little to nothing, otherwise we'd call it a job.

Granted this is an imperfect world where people need money for their necessities..

So is there a moral reason to (financially) contribute to the project to (help) make it financially sustainable? 

Maybe. Probably. Certainly it would've helped my project in the past to survive..

However, language matters a lot here. People are not purchasing copies of the software. They're certainly not under any legal obligation to do so.

Whenever a developer creates an open source project I am always happy and in awe. They are volunteering their time and effort to do so.

However just like the user has no rights to demand anything from the developer, creating the project under the AGPL for example also doesn't magically allow the developer to suddenly demand anything from the user.

It would be a very bad thing in my opinion for a project that was given away freely to the public to suddenly start trying to guilt trip users into paying. Just because of weird naming and wording.

**Comment by jordanlambrecht** [comment]
Could you do an optional monthly subscription model ranging from $0 to infinity? There's no shame badges if you pay 0, but if you do pay money you get a cool supporter icon somewhere.

**Reply by Wolfhound905** [reply]
Badges usually are meant to be shown off. I don't quite see how a badge would be nice on a self-hosted copy. I can see how badges for things like signal can have some sort of status behind them.

**Reply by stephen304** [reply]
Well since there isn't really a social aspect to immich it would be very hard to make the individual-level badges shown off anyway, so it's really just for you to feel good about. But the server-wide badge would show all your users that the server owner contributed to immich, which is one way it can be shown off if they make a different name for the server-wide badge (Server Supporter? Super Supporter?)

**Reply by InventorB** [reply]
I like the idea of a custom IOS app icon once connected to a paid server (preferably including a gold/platinum/fancy color)

**Comment by jordanlambrecht** [comment]
Also, I think it's okay to try out different revenue streams to see what works and what doesn't. Just be transparent about it. "Our team needs $X amount of dollars to keep the lights on and live comfortably. We tried out Y revenue stream for 2 months and came in over/under by Z%"

**Comment by eigenphase** [comment]
Thank you, I really love to see you listen to your community!

I really hope the outcome will be something along supporters, badges etc.

If you really want additional "income stream" down the line, consider a model like Home Assistant where you provide extra service (e.g. hosted version) at a small subscription fee.

**Reply by Waffleophagus** [reply]
I believe that's in the works too! They mentioned it in the conversation when joining FUTO!

**Comment by Snyaify** [comment]
@alextran1502 Alex, how we can support you and your team from Russia? Is any way to buy license with crypto coins?

**Reply by MMK21Hub** [reply]
Apparently FUTO is working on alternative payment methods: https://github.com/immich-app/immich/discussions/11186#discussioncomment-10086974

**Comment by Waffleophagus** [comment]
Ok, I've thought a lot about this, and hear me out. I've put a lot of thought into this. I get what you guys are trying to do, and I'm totally onboard. I love Immich and I am totally going to be supporting as soon as I'm done with this post. I get that you want to do two things simultaneously have free and open software, but also make it to where you can sustainably work on this in a meaningful way and have people who have means give you something for the value you provide. We have 20+ years of "donations" not cutting it for OSS developers in the past. I totally get wanting to instill some level obligation to want to have folks that genuinely get value out of it give some value to you guys. So my idea (that is probably terrible? Its a silly idea, I'm ok with being wrong): Keep the term License, but define it more finely. Its not a software license, be very explicit, it doesn't enable you to anything other than using Immich guilt free. Its a license to feel completely happy to use it and feel great using it. You absolutely never say that not paying anything should feel guilty, but just call it "the guilt free license."

I'm totally down with this being a bad idea and I keep flip-flopping, I definitely get the idea that, much like the discussion of the term "Open Source" and how FUTO attempted to make their own definition and how that was problematic, and the idea of a "software license" definitely has an implicit meaning within the industry as well. I just really liked the idea of a "License to be guilt free" and wanted to get the idea into the world.

**Comment by solankimihir** [comment]
Absolutely no apologies needed. Thanks for hearing us out.

**Comment by tbabej** [comment]
Contrary to others, I don't think "Supporter" or "Donate" are going to elicit same response as a "Licenced/Unlicenced". As I alluded on the [Reddit thread](https://old.reddit.com/r/selfhosted/comments/1e6hgha/immich_introduces_paid_licensing_options_unpaid/ldz0wj4/):

> It's a matter of expectations. Calling it a licence creates an expectation that you should buy it to be a good OSS citizen, calling it a donation creates an expectation that it's completely optional (i.e. you donate if you're particularly generous/thankful).

That said I see how using the same term for the license for the source and licence for the product can be confusing. Therefore I'd suggest replacing "Licenced / Unlicenced" with "Purchased / Unpurchased".

**Reply by stephen304** [reply]
> calling it a donation creates an expectation that it's completely optional

Exactly, which is why we should not make it completely optional and call it a license. If it's completely optional it should be called a donation.

**Reply by tbabej** [reply]
I can't speak for the immich team, but my understanding is that in order to make the development sustainable, the intention is to create an expectation that users *should* pay if they want to use immich. Now, they are not obligated to pay, and the code base is still open source, they have all the features, but the implicit social contract between the development team and the users is that users should pay, especially if they are getting value out of the project long term.

Donation model does not create this implicit social contract. It would be nice if it did, but I think the evidence is overwhelming that it does not.

In this sense an analogy to this situation would be tipping in restaurants in the US. If tips were called "completely optional donations to staff", the tipping rates would bu much lower. People tip mostly because they know they are expected to.

**Reply by stephen304** [reply]
I don't think comparing this to tipping is doing it the favor you think it is.

Both this and tipping are similar in that they're technically not required but social norms and expectations push you to do so. The difference is that there is no expected benefit to tipping whereas the norm for licenses is that it unlocks previously blocked features and limits.

So overall I think calling this a tip would probably be much more favorable, because it lines up more with the reality that there is no benefit to doing so.

At the same time, people hate tipping for similar reasons that they hate this license - it abuses social norms to push people to give, and this expectation is codified and evidenced by the lower wages that tipped employees receive - in expectation of tipps filling the gap. Which is why I believe tipping should be abolished as well as calling the donation a "license". Both are deceptive for different reasons. Tipping because it's a way to hide part of listed prices, and this "license" because people are deceived (intentionally or not) that the license is needed to avoid restrictions. And immich team doesn't need to do anything to be deceptive besides using the term license - their intent doesn't really matter because the term license will mislead people even if all the marketing makes it clear that there is no benefit.

And I think the immich team is maybe on the cusp of understanding that "license" is deceptive - I mean really think about it. Is the only reason people are more likely to buy a license just because it feels fair? Or could it be that having a license makes users feel safe against restrictions on the software - restrictions that supposedly are not coming? I think it's a little bit of both, and benefitting from giving users that feeling of safety when the threat (feature restrictions and limitations) does not exist is plainly speaking deceptive.

Again I want to reiterate that I fully believe the Immich team has good intentions, but I do see why people were rubbed the wrong way or felt deceived even if there was no intention of that.

**Reply by mvdkleijn** [reply]
> I can't speak for the immich team, but my understanding is that in order to make the development sustainable, the intention is to create an expectation that users _should_ pay if they want to use immich. Now, they are not obligated to pay, and the code base is still open source, they have all the features, but the implicit social contract between the development team and the users is that users should pay, especially if they are getting value out of the project long term.
> 
> Donation model does not create this implicit social contract. It would be nice if it did, but I think the evidence is overwhelming that it does not.
> 
> In this sense an analogy to this situation would be tipping in restaurants in the US. If tips were called "completely optional donations to staff", the tipping rates would bu much lower. People tip mostly because they know they are expected to.

I think comparing this to the tipping situation in the USA is not healthy since that's a screwed up system to begin with.

**Comment by qbuns** [comment]
I think labeling it as suggested donation or suggested support is more accurate. similarly to terminology museums and non-profits use.

**Comment by sannidhyaroy** [comment]
## I'm not a professional designer, but I hope these attached screenshots get the basic idea right:

<img width="270" alt="Immich Sidebar" src="https://github.com/user-attachments/assets/43a66d2f-04d1-43a8-82d9-2b13d8abba3d">
<img width="1440" alt="Immich Support Popup" src="https://github.com/user-attachments/assets/eabf8324-f77e-4602-8e6a-c8b76674f4a0">

**Reply by Blisterexe** [reply]
![image](https://github.com/user-attachments/assets/df1b05a1-ee2b-4c30-b6cf-6b1685c715c5)
tried my hand at making a better-looking mockup

**Reply by erkexzcx** [reply]
That's really neat!

**Reply by InventorB** [reply]
I like that.

**Reply by basdp** [reply]
this is the correct solution to this whole controversy.

**Reply by Lanhild** [reply]
Best way to implement it

**Comment by diyoyo** [comment]
Hi, my 2 cents: I don't care about the wording, but the extra space it takes is making the scrollbar of the left pane visible, and imho, it ruins the overall design. 
yes, I have the Bookmarks bar always visible on my browser, and I shouldn't have to make a choice ;)

<img width="454" alt="image" src="https://github.com/user-attachments/assets/eaad7035-06d8-4353-935f-ff4e39fa94c1">

**Reply by alextran1502** [reply]
What is this your screensize you normally use for Immich, in pixel?

**Reply by stephen304** [reply]
My screen is 1080px tall and I get a few pixels of scroll too.

**Reply by diyoyo** [reply]
I think it's 1680x1050, on macbook pro 13in 2018

**Comment by jennilicious** [comment]
@alextran1502 I want to personally thank you and the rest and the team for listening. I still initially purchased a "license" the first day because I ultimately want Immich to succeed, despite the controversy of the of the term "license." Please understand that that the frustration from the community stems from fear of a product they all loved, from experiences of other open source projects.

I do agree with the other folks for 'community' or 'supporter' terms, as it carries a warm meaning. I want to admit that when v109 first rolled out, it did take me a LONG while to decide, as I felt conflicted if I should commit due to negative connotations of the term "license." I was not sure if someone would end of forking a new project, or the future would lead to paywalled features. Ultimately after reading the team's comments on previous discussions, I went forward as I felt confident enough that the team will keep their word.

Just wanted to share what I felt during that moment, as I fear those discovering Immich for the first time may experience the same feelings, turning away to other projects. It is heartwarming to see this announcement. To carry on its success, this is something we must do to ensure the project continues to grow new folks, winning their trust for them to continue advocating this wonderful project.

Cheers!

**Comment by Arimodu** [comment]
I personally dont have a problem with the wording of "paying" for software instead of "donating". What I dont like is the "Unlicensed" badge being always visible (also the wording for Unlicensed is not great).

If you went with a "Buy immich" button in the user popout I think the response wouldve been much better in general. If you went with that you couldve done the same thing the FUTO keyboard and Grayjay do and to be honest, Im fully on board with that. This goes back to giving the impression "this is a paid app, not a free one" but also doesnt shove not paying in your face. As a sidenote, I also like the "I already paid" button on the other FUTO apps. As Louis Rossmann says, "clicking that button and not paying, thats between you and your god". 

And as a sidenote, before people start throwing stones at me telling me that I would just use that button to be a scumbag, then yes, I paid for FUTO keyboard. And I would pay for immich as well in that arrangement.

I dont want free software over sustainable software. Donations have proved themselves unreliable time and time again for sustainability, while paid software has been doing just fine.

Anyway, thank you for listening to the community!

**Comment by crtvrmn** [comment]
if you sell license (or whatever you will call it) keys, be prepared for the army of normies that will message you every day because they have lost their license key.

some quick suggestions: 

- count the times or amount of payments, offer a optional (!) badge to show there. if i would give you money, just to look at clutter id be pissed and rather spend the time on blocking that thing or forking the project without the annoyance. 
- use a patreon-like-system instead
- sell a encrypted hosted web-service for normies (im not sure why anyone would buy this, but people are strange) instead of "keys"

**Comment by martimors** [comment]
I think this makes sense as a whole. Getting up and running with Immich, including migrating the entire household from google, took a lot of time for me, and it was never certain it would succeed. The intention all along was to give back by setting up a recurring monthly or yearly donation to the project. Instead of a one-time 100 USD fee, I'd personally prefer a perpetual monthly donation, but I'm fine with holding that money for some months until it amounts to the whole 100 USD and just paying then.

**Comment by Deedikjupijn** [comment]
Maybe they could actually make the license mean something, for example buying yourself into an agreement with futo that legally binds them to the promise of keeping the software open source/free for a certain time period (a couple years from purchase), this would make sure their meaningless promises actually start to matter.

Not sure how these kinds of things work when taking in consideration different country's and laws but im sure this would not actually be the hard part.

Really curious what your/others take on this is and as always hope you have a great day!

**Comment by monotok** [comment]
@alextran1502 I currently sponsor yourself. Do we get a "license" as an existing sponsor? I don't mind purchasing a lifetime license instead but was planning on increasing my sponsorship in due course. Immich really is fantastic software, once it has https://github.com/immich-app/immich/issues/165 then it will be perfect for my needs. Thanks for all your hard work!

**Reply by alextran1502** [reply]
Yes, we are working on a mechanism where you can claim that automatically
